hive/operation.go

```diff
@@ -67,10 +67,27 @@ func (op *Operation) FetchResults(ctx context.Context, schema *TableSchema) (*Re
 	if err != nil {
 		return nil, err
 	}
+	if ctx.Done() != nil {
+		cnt := 0
+		for resp.Results == nil {
+			select {
+			case <-ctx.Done():
+				break
+			default:
+				resp, err = fetch(ctx, op, schema)
+				cnt++
+			}
+			op.hive.log.Printf("refetch time %d\n", cnt)
+		}
+	}
+	resultLen := 0
+	if resp.Results != nil {
```

Comment:
minor suggestion. we can move nil check into length function to make code more "go-way".

Reply:
good suggestion.

Reply:
please merge the request, thx!

Reply:
Thanks for fixing the issue.